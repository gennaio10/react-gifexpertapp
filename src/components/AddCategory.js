import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleOnSubmit = (e) => {
    //evita que se recarga la pagina
    e.preventDefault();

    if (inputValue.trim().length > 1) {
      setCategories((categoriesE) => [inputValue, ...categoriesE]);
      setInputValue("");
    }
  };
  
  return (
    <form onSubmit={handleOnSubmit}>
      <h2>Add Category</h2>
      <input id="icategoria" type="text" value={inputValue} onChange={handleChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
