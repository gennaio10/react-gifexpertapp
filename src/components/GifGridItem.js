import React from "react";
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
  return (
    <div id="dimg" className="card animate__animated animate__bounce">
      <img id="isrc" src={url} alt={title} />
      <p id="ptitle">{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};



