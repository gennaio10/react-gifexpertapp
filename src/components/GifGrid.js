import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data:imagenes, loadind } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loadind && <p>Loading...</p>}
      <div className="card-grid">
        {imagenes.map((imagenConUrl) => (
          <GifGridItem key={imagenConUrl.id} {...imagenConUrl} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
