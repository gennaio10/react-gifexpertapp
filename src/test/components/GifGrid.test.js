import React from "react";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";

// simula que se ejecuto y se controla la informacion que se supone va a responder
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas da <GifGrid />", () => {
  const category = "Goku";

  test("debe de mostrar el componenete correctamente", () => {
    useFetchGifs.mockReturnValue({ data: [], loadind: true });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    // hacer como si hubiera pasado por el custom hook y hubiera decuelto informacion
    const gifs = [
      {
        id: "ABC",
        title: "https://localhost/algo.jpg",
        url: "Un titulo",
      },
      {
        id: "ABC123",
        title: "https://localhost/algo.jpg",
        url: "Un titulo",
      },
    ];
    useFetchGifs.mockReturnValue({ data: gifs, loadind: false });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
