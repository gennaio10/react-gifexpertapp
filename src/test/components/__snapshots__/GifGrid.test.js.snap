// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas da <GifGrid /> debe de mostrar el componenete correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas da <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="https://localhost/algo.jpg"
      url="Un titulo"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="https://localhost/algo.jpg"
      url="Un titulo"
    />
  </div>
</Fragment>
`;
